t/plugin/aws-lambda.t

```diff
@@ -0,0 +1,299 @@
+                        secretkey = "key2"
+                    }
+                }
+            })
+            if not ok then
+                ngx.say(err)
+            else
+                ngx.say("done")
+            end
+        }
+    }
+--- response_body
+property "authorization" validation failed: property "iam" validation failed: property "accesskey" is required
+
+
+
+=== TEST 3: create route with aws plugin enabled
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+
+            local code, body = t('/apisix/admin/routes/1',
+                 ngx.HTTP_PUT,
+                 [[{
+                        "plugins": {
+                            "aws-lambda": {
+                                "function_uri": "http://localhost:8765/httptrigger",
+                                "authorization": {
+                                    "apikey" : "testkey"
+                                }
+                            }
+                        },
+                        "uri": "/aws"
+                }]],
+                [[{
+                    "node": {
+                        "value": {
+                            "plugins": {
+                                "aws-lambda": {
+                                    "keepalive": true,
+                                    "timeout": 3000,
+                                    "ssl_verify": true,
+                                    "keepalive_timeout": 60000,
+                                    "keepalive_pool": 5,
+                                    "function_uri": "http://localhost:8765/httptrigger",
+                                    "authorization": {
+                                        "apikey": "testkey"
+                                    }
+                                }
+                            },
+                            "uri": "/aws"
+                        },
+                        "key": "/apisix/routes/1"
+                    },
+                    "action": "set"
+                }]]
+                )
+
+            if code >= 300 then
+                ngx.status = code
+                ngx.say("fail")
+                return
+            end
+
+            ngx.say(body)
+        }
+    }
+--- response_body
+passed
+
+
+
+=== TEST 4: Test plugin endpoint
+--- config
+    location /t {
+        content_by_lua_block {
+            local t = require("lib.test_admin").test
+            local core = require("apisix.core")
+
+            local code, _, body, headers = t("/aws", "GET")
+             if code >= 300 then
+                ngx.status = code
+                ngx.say(body)
+                return
+            end
+
+            -- headers proxied 2 times -- one by plugin, another by this test case
+            core.response.set_header(headers)
+            ngx.print(body)
+        }
+    }
+--- response_body
+aws lambda invoked
+--- response_headers
+Content-Length: 19
```

Comment:
why check the `Content-Length`

Reply:
Hi, thanks for the review.
It just checks if the headers that are being received from upstream lambda are being forwarded by the plugin. Not just `Content-Length` specifically but all headers from the mock HTTP endpoint